core/uemis-downloader.c

```diff
@@ -1117,7 +1126,32 @@ static bool load_uemis_divespot(const char *mountpath, int divespot_id)
 static void get_uemis_divespot(const char *mountpath, int divespot_id, struct dive *dive)
 {
 	struct dive_site *nds = get_dive_site_by_uuid(dive->dive_site_uuid);
-	if (nds && nds->name && strstr(nds->name,"from Uemis")) {
+	struct divespot_downloads **pdl = &divespot_downloads;
+	struct divespot_downloads *dl = *pdl;
+	bool already_downloaded = false;
+
+	/*
+	 * Search linked list of already downloaded
+	 * divespots using the uemis's divespot_id
+	 */
+	while(dl && !already_downloaded)
+	{
+		if(dl->divespot_id == divespot_id)
+		{
+			/*
+			 * This divespot_id was already requested.
+			 */
```

Comment:
please reduce this comment to a single line:
` /* This divespot_id was already requested. */ `